### PR TITLE
fix: Allow updating state as node end

### PR DESCRIPTION
**Pull request by bracesproul**

### CODE REVIEW

File: libs/langgraph/src/pregel/index.ts [hunk 1]
```diff
@@ -642,8 +646,13 @@ export class Pregel<
     let checkpointConfig = patchConfigurable(config, {
       checkpoint_ns: config.configurable?.checkpoint_ns ?? "",
     });
+    let checkpointMetadata = config.metadata ?? {};
     if (saved?.config.configurable) {
       checkpointConfig = patchConfigurable(config, saved.config.configurable);
+      checkpointMetadata = {
+        ...saved.metadata,
+        ...checkpointMetadata,
+      };
```
Comment (bracesproul) [inline libs/langgraph/src/pregel/index.ts:655]: is in PY, was missing in JS

File: libs/langgraph/src/pregel/index.ts [hunk 2]
```diff
@@ -661,6 +670,79 @@ export class Pregel<
...
+      if (saved) {
+        // tasks for this checkpoint
+        const nextTasks = _prepareNextTasks(
+          checkpoint,
+          saved.pendingWrites || [],
+          this.nodes,
+          channels,
+          managed,
+          saved.config,
+          true, // Should I pass forExecution=true here?
```
Comment (bracesproul) [inline libs/langgraph/src/pregel/index.ts:693]: question

Comment (nfcampos) [reply libs/langgraph/src/pregel/index.ts:693]: yes

**Review by nfcampos — DISMISSED** [review]

File: libs/langgraph/src/pregel/index.ts (hunk 2 above)
Comment (bracesproul) [inline libs/langgraph/src/pregel/index.ts:693]:
```suggestion
          true,
```

**Pull request merged**